dev/build-libs.js

```diff
@@ -64,6 +64,7 @@ export async function buildLibs() {
     const schemaFileNames = fs.readdirSync(schemaDir);
     const schemas = schemaFileNames.map((schemaFileName) => {
         /** @type {import('ajv').AnySchema} */
+        // eslint-disable-next-line sonarjs/prefer-immediate-return
```

Comment:
There are a few cases where I had to disable `sonarjs/prefer-immediate-return` because it needs the type hint. I could have used a cast, but a type hint is a bit safer and the syntax a bit cleaner.

Reply:
Maybe we should open an upstream issue for this case.

Reply:
https://github.com/SonarSource/eslint-plugin-sonarjs/issues/448

Reply:
I'm fine with either way. It's kind of an edge case and idk if sonarjs is trying to account type-annotated JS in any of their codebase. In general, type-annotated JS is kinda a weird middle-ground between JS and TS that I wouldn't be surprised that people wouldn't be explicitly supporting.

Since this use case is pretty infrequent in the codebase so far, personally I wouldn't be too upset by having to do it the occasional time.

That said, I have no problem with an upstream issue like you created, so we can see if they follow up on it at all.